services/cd-service/pkg/service/git_test.go

```diff
@@ -393,6 +395,32 @@ func TestGetCommitInfo(t *testing.T) {
 			expectedError:    status.Error(codes.NotFound, "error: commit bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb was not found in the manifest repo"),
 			expectedResponse: nil,
 		},
+		{
+			name: "find a commit by prefix",
+			transformers: []rp.Transformer{
+				&rp.CreateApplicationVersion{
+					Application:    "app",
+					SourceCommitId: "32a5b7b27fe0e7c328e8ec4615cb34750bc328bd",
+					SourceMessage:  "some message",
+				},
+			},
+			request: &api.GetCommitInfoRequest{
+				CommitHash: "32a5b7b27",
+			},
+			expectedResponse: &api.GetCommitInfoResponse{
+				CommitHash:    "32a5b7b27fe0e7c328e8ec4615cb34750bc328bd",
+				CommitMessage: "some message",
+				TouchedApps:   []string{"app"},
+				Events: []*api.Event{
+					{
+						CreatedAt: timestamppb.New(fixedTime()),
+						EventType: &api.Event_CreateReleaseEvent{
+							CreateReleaseEvent: &api.CreateReleaseEvent{},
+						},
+					},
+				},
+			},
+		},
```

Comment:
Perhaps we need to add test for the case when the commit ID length is less than 7 and greater than 40?

Reply:
I don't think that those are especially important cases. Even if we would accidentally give out a commit if the length is <7 , there's no harm in it.